### ISSUE TITLE
Firefox: You cannot use speech synthesis because the Speech Dispatcher library is missing.

### ISSUE THREAD

**Issue by Ashvith10** [NixOS/nixpkgs#248179]
### Describe the bug
Speech Dispatcher library is missing for Firefox, which doesn't look like the case as it is still present here:

https://github.com/NixOS/nixpkgs/blob/011567f35433879aae5024fc6ec53f2a0568a6c4/pkgs/applications/networking/browsers/firefox/wrapper.nix#L19

### Steps To Reproduce
1. Launch Firefox
2. Open a web-app that uses Speech Dispatcher.
3. Ribbon bar with warning will pop-up

### Expected behavior
Speech Dispatcher should work on Firefox

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/113123021/fc10bc55-7752-49a7-bb6e-7a84b0f2a007)

### Additional context
Probably not relevant, but when I launch Firefox from the terminal, here are the logs:
```
$ firefox 
[GFX1-]: vaapitest: ERROR
[GFX1-]: vaapitest: VA-API test failed: failed to initialise VAAPI connection.

[GFX1-]: Wayland protocol error: warning: queue 7f8f6b0fcb20 destroyed while proxies still attached:

[GFX1-]: Wayland protocol error:   wl_registry@47 still attached

[GFX1-]: Wayland protocol error: warning: queue 7f8f6b003fd0 destroyed while proxies still attached:

[GFX1-]: Wayland protocol error:   wl_registry@65 still attached
```

### Notify maintainers
@lovesegfault 
@mweinelt 
(last three contributors to Firefox `wrapper.nix`)
@flokli
@amjoseph-nixpkgs 
@austinbutler 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.43, NixOS, 23.05 (Stoat), 23.05.2613.011567f35433`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(ashvith): `""`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by mweinelt** [comment]
Can you try the option mentioned here?

https://github.com/NixOS/nixpkgs/pull/210453#issuecomment-1410035331

**Comment by Ashvith10** [comment]
@mweinelt It works now. I was wondering about why it was disabled by default? Looks like we might have to create a new issue to improve the [Firefox wiki](https://nixos.wiki/wiki/Firefox).

**Comment by towards-a-new-leftypol** [comment]
+1

**Comment by Nyabinary** [comment]
This issue should be closed if the problem is resolved.

**Comment by mweinelt** [comment]
We should probably look into enabling it by default.

**Comment by Ashvith10** [comment]
@Nyabinary yes, the problem was resolved, but since this isnt documented, I left it open. Perhaps, we might need to close this, and open a new issue with the intention to improve the documentation.

@mweinelt I agree with you. It makes no sense disabling an accessibility feature by default. This is something that is very handy, in case if you have a different user trying to use your device.

**Comment by Nyabinary** [comment]
I agree but lets make this a different issue so the title can be more accurate and we can have more discussion about it.

**Comment by vcunat** [comment]
Default: speechd brings in about 0.7G data IIRC.

**Comment by Ashvith10** [comment]
@Nyabinary I'm closing this issue. Opened #253258 to discuss about documenting speech dispatcher for Firefox.
